File: src/main/java/nz/co/searchwellington/model/PublishedResourceImpl.java
package nz.co.searchwellington.model;

public abstract class PublishedResourceImpl extends ResourceImpl implements PublishedResource {
	
	private static final long serialVersionUID = 1L;
	
	protected Website publisher;

    public Website getPublisher() {
        return publisher;
    }

    public void setPublisher(Website publisher) {
        this.publisher = publisher;
    }

	public String getPublisherName() {
		if (publisher != null) {
			return publisher.getName();
		}
		return null;
	}
	
}

File: src/main/scala/nz/co/searchwellington/model/PublishedResourceImpl.scala
package nz.co.searchwellington.model

@SerialVersionUID(1L)
abstract class PublishedResourceImpl extends Resource with PublishedResource {
  protected var publisher: Website = null

  override def getPublisher: Website = publisher

  override def setPublisher(publisher: Website): Unit = this.publisher = publisher

  override def getPublisherName: String = {
    if (publisher != null) return publisher.getName
    null
  }
}
